jib-core/src/main/java/com/google/cloud/tools/jib/configuration/DockerHealthCheck.java

```diff
@@ -0,0 +1,184 @@
+     * Sets the number of times to retry the healthcheck before the container is considered to be
+     * unhealthy.
+     *
+     * @param retries the number of retries before the container is considered to be unhealthy
+     * @return this
+     */
+    public Builder setRetries(int retries) {
+      this.retries = retries;
+      return this;
+    }
+
+    public DockerHealthCheck build() {
+      return new DockerHealthCheck(command, interval, timeout, startPeriod, retries);
+    }
+  }
+
+  /**
+   * Creates a disabled {@link DockerHealthCheck}.
+   *
+   * @return the new {@link DockerHealthCheck}
+   */
+  public static DockerHealthCheck disabled() {
+    return new DockerHealthCheck(ImmutableList.of("NONE"), null, null, null, null);
+  }
+
+  /**
+   * Creates a new {@link DockerHealthCheck.Builder} with the command set to be inherited from the
+   * base image.
+   *
+   * @return the new {@link DockerHealthCheck.Builder}
+   */
+  public static Builder withInheritedCommand() {
+    return new Builder(ImmutableList.of());
+  }
+
+  /**
+   * Creates a new {@link DockerHealthCheck.Builder} with the specified healthcheck command to be
+   * directly executed.
+   *
+   * @param command the healthcheck command to execute
+   * @return the new {@link DockerHealthCheck.Builder}
+   */
+  public static Builder withExecCommand(List<String> command) {
+    return new Builder(ImmutableList.<String>builder().add("CMD").addAll(command).build());
+  }
+
+  /**
+   * Creates a new {@link DockerHealthCheck.Builder} with the specified healthcheck command to be
+   * directly executed.
+   *
+   * @param command the healthcheck command to execute
+   * @return the new {@link DockerHealthCheck.Builder}
+   */
+  public static Builder withExecCommand(String... command) {
+    return new Builder(
+        ImmutableList.<String>builder().add("CMD").addAll(Arrays.asList(command)).build());
+  }
+
+  /**
+   * Creates a new {@link DockerHealthCheck.Builder} with the specified healthcheck command to be
+   * run by the container's default shell. This command cannot be run on containers with no default
+   * shell.
+   *
+   * @param command the shell command to run
+   * @return the new {@link DockerHealthCheck.Builder}
+   */
+  public static Builder withShellCommand(String command) {
+    return new Builder(ImmutableList.of("CMD-SHELL", command));
+  }
+
+  private final ImmutableList<String> command;
+  @Nullable private final Duration interval;
+  @Nullable private final Duration timeout;
+  @Nullable private final Duration startPeriod;
+  @Nullable private final Integer retries;
+
+  private DockerHealthCheck(
+      ImmutableList<String> command,
+      @Nullable Duration interval,
+      @Nullable Duration timeout,
+      @Nullable Duration startPeriod,
+      @Nullable Integer retries) {
+    this.command = command;
+    this.interval = interval;
+    this.timeout = timeout;
+    this.startPeriod = startPeriod;
+    this.retries = retries;
+  }
+
```

Comment:
Should we add a `shouldInherit()` method that indicates that the healthcheck should inherit rather than having the caller check the `getCommand` is an empty list?